pagopa/interop-be-monorepo
# IMN-344 - Implementing SQS queue manager

Closes [IMN-344](https://pagopa.atlassian.net/browse/IMN-344)

## 💡 Looks big, but most of the diff is in the `pnpm-lock`

This PR implements an SQS queue writer, based on the one from the Scala codebase: [see here](https://github.com/pagopa/interop-commons/blob/1f4fa7b28b005e2af518c4abe3139daee8ac7fee/queue-manager/src/main/scala/it/pagopa/interop/commons/queue/impl/SQSWriter.scala#L25)

It will then be used to send notification messages to the existing SQS queue, as done by the `catalog-management` service in the Scala codebase: [see here](https://github.com/pagopa/interop-be-catalog-management/blob/58f22eef590b2bc267cb7fbf8777237d0eae3b23/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/projection/CatalogNotificationProjection.scala#L27) (tracked as a further task in Jira)

[IMN-344]: https://pagopa.atlassian.net/browse/IMN-344?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

I also set up an ElasticMQ instance in the local dev env: ElasticMQ is an in-memory message queue system with an AWS SQS-compatible API: https://github.com/softwaremill/elasticmq

# Tests

## Automated tests ✅
I wrote automated integration tests that perform a send and a receive on an ElasticMQ queue (setup using `testcontainers`)

<img width="754" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/a4e4f360-ed2e-4ee4-b944-91737c872cb0">

## Manual tests ✅

Pointing the automated tests suite to the address of the local instance of ElasticMQ, instead of sending messages into the container setup by `testcontainers`, I can see the message in the queue from the ElasticMQ UI:

### Before sending
<img width="1131" alt="Screenshot 2024-03-13 at 15 32 03" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/91f6a4b7-237d-4375-9e23-0556d8aefa0d">

### After sending
<img width="994" alt="Screenshot 2024-03-13 at 15 33 32" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/f78dff3a-7350-4564-b355-1401d60e87c2">
